datafusion/core/tests/user_defined/user_defined_scalar_functions.rs

```diff
@@ -655,6 +655,7 @@ impl ScalarUDFImpl for TakeUDF {
         &self,
         arg_exprs: &[Expr],
         schema: &dyn ExprSchema,
+        _arg_data_types: &[DataType],
```

Comment:
seems unused 🤔

Reply:
I think it's right for usage, for `return_type_from_exprs` user would calculate the return_type form `Expr` and `Schema`. And for a user defined `return_type_from_exprs`, it would not re-calculate the `arg_data_types` from `args`.

Reply:
if `_arg_data_types` is not provided, we calculate them from `args` and `schema`. They are mutual exclusive.

Reply:
(note this is a test)